libs/platform/core/src/orchestration/app/modular-app-builder.ts

```diff
@@ -106,21 +102,19 @@ export class ModularAppBuilder extends SimpleAppBuilder<AppBuilderWithModules> {
 
   async create(): Promise<App> {
     if (this.providers.length) {
-      this.plugins.unshift(
-        new InjectionPlugin(this.providers, this.options?.injector)
-      );
+      this.with(new InjectionPlugin(this.providers, this.options?.injector));
     }
 
     if (this.resources) {
-      this.plugins.unshift(new ResourcePlugin(this.resources));
+      this.with(new ResourcePlugin(this.resources));
```

Comment:
We wanted that these plugins have first order of initialization

Reply:
But it's not correct to rely on the order of plugin registration, we have special hooks for that (before/after apply) which should be used instead to establish correct order.
I will have a look into those plugins to correct the order but we should not couple registration order with execution order.